setup-yamlfmt/action.yaml

```diff
@@ -0,0 +1,44 @@
+# SPDX-FileCopyrightText: Copyright 2023 Chainguard, Inc.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Shamelessly stolen from https://github.com/imjasonh/setup-ko
+
+name: 'Setup yamlfmt'
+description: 'Install yamlfmt'
+
+inputs:
+  version:
+    description: |
+      Version to install (tip, latest, or a tag eg "v0.6.0")
+    required: false
+    default: "latest"
+
+runs:
+  using: "composite"
+  steps:
+    - shell: bash
+      name: Install yamlfmt binary
+      run: |
+        set -ex
+        # - if version is "tip", install from tip of main.
+        # - if version is "latest", look up latest release.
+        # - otherwise, install the specified version.
+        case ${{ inputs.version }} in
+        tip)
+          echo "Installing yamlfmt using go install"
+          go install github.com/google/yamlfmt@main
+          ;;
+        latest)
+          tag=$(curl -L -s -u "username:${{ github.token }}" https://api.github.com/repos/google/yamlfmt/releases/latest | jq -r '.tag_name')
+          ;;
+        *)
+          tag="${{ inputs.version }}"
+        esac
```

Comment:
I _think_ the `go install <repo>@<git-ref>` syntax handles `main` and `latest` automatically, in addition to tagged versions. Would it make sense to let the `go install` command handle the version parsing here?